### PR TITLE
Protocol and metamodel changes for Mastery Runtime

### PR DESCRIPTION
**Pull request by agshubh-97**
## How did you test this change?

- [ ] Test(s) added
- [ ] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: .changeset/big-tables-hang.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-extension-dsl-mastery': patch
+---
+
+Protocol and Metamodel changes for Mastery Runtime
```
Comment (gayathrir11) [inline .changeset/big-tables-hang.md:5]: Can you please modify this `Add support for Mastery Runtime`

File: packages/legend-extension-dsl-mastery/src/graph-manager/protocol/pure/v1/model/packageableElements/mastery/V1_DSL_Mastery_Runtime.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Hashable, hashArray } from '@finos/legend-shared';
+import { MASTERY_HASH_STRUCTURE } from '../../../../../../../graph/DSL_Mastery_HashUtils';
+import { V1_PackageableElement } from '@finos/legend-graph';
+
+export abstract class V1_Runtime
+  extends V1_PackageableElement
```
Comment (gayathrir11) [inline packages/legend-extension-dsl-mastery/src/graph-manager/protocol/pure/v1/model/packageableElements/mastery/V1_DSL_Mastery_Runtime.ts:22]: Isn't it called `MasteryRuntime`? If that's the case then can you please rename it to `MasteryRuntime` everywhere

File: packages/legend-extension-dsl-mastery/src/graph-manager/protocol/pure/v1/transformation/pureProtocol/V1_DSL_Mastery_ProtocolHelper.ts [hunk 3]
```diff
@@ -925,3 +926,50 @@ export const V1_masterRecordDefinitionModelSchema = (
...
+      ).V1_getExtraRuntimeProtocolSerializers?.() ?? [],
+  );
+  for (const serializer of extraRuntimeSerializers) {
+    const runtimeJson = serializer(protocol, plugins);
+    if (runtimeJson) {
+      return runtimeJson;
+    }
+  }
+  throw new UnsupportedOperationError(
+    `Can't serialize runtime Json: no compatible serializer available from plugins`,
```
Comment (gayathrir11) [inline packages/legend-extension-dsl-mastery/src/graph-manager/protocol/pure/v1/transformation/pureProtocol/V1_DSL_Mastery_ProtocolHelper.ts:951]: Please modify this to `Can't serialize mastery runtime: no compatible serializer available from plugins

**Review by gayathrir11 — CHANGES_REQUESTED** [review]
Please fix the pipelines and also some tests in `legend-manual-tests` and DSL_Mastery_Roundtrip.test.ts

**Comment by gayathrir11** [comment]
I don't think you have made changes to `DSL_Mastery_PureProtocolProcessorPlugin.ts`

File: packages/legend-extension-dsl-mastery/src/graph-manager/protocol/pure/v1/model/packageableElements/mastery/V1_DSL_Mastery_Runtime.ts (hunk 2 above)
Comment (agshubh-97) [reply packages/legend-extension-dsl-mastery/src/graph-manager/protocol/pure/v1/model/packageableElements/mastery/V1_DSL_Mastery_Runtime.ts:22]: The convention that we've used in the rest of the protocols is the same as this one. So, stuck to V1_Runtime. Shall I switch to MasteryRuntime?

**Comment by agshubh-97** [comment]
> I don't think you have made changes to `DSL_Mastery_PureProtocolProcessorPlugin.ts`

I've added changes in DSL_Mastery_PureProtocolProcessorPlugin_Extension as Runtime is abstract and did not have any implementations. Please correct me if I'm missing something.

File: packages/legend-extension-dsl-mastery/src/graph-manager/protocol/pure/v1/model/packageableElements/mastery/V1_DSL_Mastery_Runtime.ts (hunk 2 above)
Comment (gayathrir11) [reply packages/legend-extension-dsl-mastery/src/graph-manager/protocol/pure/v1/model/packageableElements/mastery/V1_DSL_Mastery_Runtime.ts:22]: For protocol models can you modify it to `V1_MasteryRuntime` and for metamodels `MasteryRuntime`

**Comment by gayathrir11** [comment]
> > I don't think you have made changes to `DSL_Mastery_PureProtocolProcessorPlugin.ts`
> 
> I've added changes in DSL_Mastery_PureProtocolProcessorPlugin_Extension as Runtime is abstract and did not have any implementations. Please correct me if I'm missing something.

Please take a look at how other functions defined in the interface are getting called. You will need to do something similar so that when any element of type `MasteryRuntime` is found it can be routed to implementations of the function when you implement the plugin

**Pull request closed without merging**